src/shared/api/urls.ts

```diff
@@ -219,9 +219,11 @@ export function getDocsUrl(sourceUrl:string,docsBaseUrl?:string): string {
 }
 
 export function getWholeSlideViewerUrl(ids: string[], userName: string): string {
-    return ids.length > 1 ? `https://slides.mskcc.org/cbioportal?ids=${ids.join('.svs;') + '.svs'}&user=${userName}&annotation=off` : ids.length === 1 ? `https://slides.mskcc.org/cbioportal?ids=${ids[0] + '.svs'}&user=${userName}&annotation=off&filetree=off` : "";
+    const tokenInfo = JSON.parse(AppConfig.serverConfig.mskWholeSlideViewerToken);
+    const token = `&token=${tokenInfo.token}`;
+    const time = `&t=${tokenInfo.time}`
+    return ids.length > 1 ? `https://slides.mskcc.org/cbioportal?ids=${ids.join('.svs;') + '.svs'}&user=${userName}${time}${token}&annotation=off` : ids.length === 1 ? `https://slides.mskcc.org/cbioportal?ids=${ids[0] + '.svs'}&user=${userName}${time}${token}&annotation=off&filetree=off` : "";
```

Comment:
i think this is a nested ternary operation, which should be avoided.  

overall seems over complicated.  if first condition is > 1 then why do you also need a subsequent ===1 ?
 
how about:  
map((name)=>name+".svs").join(";")